galsim/compound.py

```diff
@@ -902,3 +904,162 @@ def __setstate__(self, d):
 _galsim.SBFourierSqrt.__setstate__ = lambda self, state: 1
 _galsim.SBFourierSqrt.__repr__ = lambda self: \
         'galsim._galsim.SBFourierSqrt(%r, %r)'%self.__getinitargs__()
+
+
+class RandomWalk(Sum):
+    """
+
+    A class for generating a set of point sources distributed using a random
+    walk.  Uses of this profile include representing an "irregular" galaxy, or
+    adding this profile to an Exponential to represent knots of star formation.
+
+    Random walk profiles have a "shape noise" of g~0.05, but can be sheared to
+    give an additional ellipticity, for example to follow that of an associated
+    disk.
+    
+    Initialization
+    --------------
+    @param  npoints  Number of point sources to generate.
+    @param  hlr      Half light radius of the distribution of points.  This
+                     is the mean half light radius produced by an 
+                     infinite number of points.  A single instance will be
+                     noisy.
+    @param  flux     Total flux in all point sources.  Default 1
+    @param  nstep    Number of steps in random walk.  Default 40
+    @param  rng      Optional random number generator. Should be a UniformDeviate.
+    @param  gsparams GSParams for the gaussians representing each point source.
+
+    Methods
+    -------
+
+    This class inherits from galsim.Sum. Additional methods, implemented as
+    read-only properties, are provided as "getters" for the basic parameters:
+
+        .npoints
+        .half_light_radius
+        .flux
+        .nstep
+        .gaussians
+            The list of galsim.Gaussian objects representing the points
+
+    Notes
+    -----
+
+    - The algorithm is a modified version of that presented in
+      https://arxiv.org/abs/1312.5514v3 : There is no outer cutoff to how far a
+      point can wander.
+    - This class inherits from galsim.Sum
```

Comment:
This note is redundant with the above statement in Methods.  Plus, I think the user shouldn't probably care about this implementation detail other than how it relates to which methods are available, so the above statement is probably sufficient.